go/otgrpc/shared.go

```diff
@@ -18,6 +20,7 @@ type metadataReaderWriter struct {
 }
 
 func (w metadataReaderWriter) Set(key, val string) {
+	key = strings.ToLower(key)
```

Comment:
is this necessary? the HTTPHeaders carriers are supposed to be case-insensitive...